src/r1cs_nark_as/data_structures.rs

```diff
@@ -1,5 +1,6 @@
 use crate::hp_as::{
     InputInstance as HPInputInstance, InputWitness as HPInputWitness, Proof as HPProof,
+    ProofHidingCommitments, ProofTCommitments,
```

Comment:
What is a `ProofTCommitment`?

Reply:
ProofTCommitment is located here: https://github.com/arkworks-rs/accumulation/blob/a5d66cf72bd5392dacacdb9caf48bb20b326452a/src/hp_as/data_structures.rs#L105. The commitments refer to step 8 of the prover in Section 7.1 (https://eprint.iacr.org/2020/1618.pdf).

Reply:
What does the `T` stand for? Maybe we should replace it with something more descriptive.

Reply:
How about `ProductPolynomialCommitment`?

Reply:
Yeah that sounds good.

Reply:
Since the refactor is a separate issue, I have opened a new PR for it. #32 

Once that PR has been merged, I will update this branch.

Reply:
Let’s rebase and merge this now